src/pcp.rs

```diff
@@ -343,17 +353,31 @@ impl<F: FieldElement> From<Vec<F>> for Proof<F> {
 
 /// Generate the verifier for an input and proof (or the verifier share for an input share and
 /// proof share).
-pub fn query<F, G, V>(x: &V, pf: &Proof<F>, rand: &[F]) -> Result<Verifier<F>, PcpError>
+///
+/// Parameters:
+/// * `x` is the input.
+/// * `pf` is the proof.
+/// * `query_rand` is the verifier's randomness.
+/// * `joint_rand` is the randomness shared by the prover and verifier.
+pub fn query<F, G, V>(
+    x: &V,
+    pf: &Proof<F>,
+    query_rand: &[F],
+    joint_rand: &[F],
+) -> Result<Verifier<F>, PcpError>
 where
     F: FieldElement,
     G: Gadget<F>,
     V: Value<F, G>,
 {
+    if query_rand.len() != 1 {
```

Comment:
If the query randomness can only ever have one element, why not make it `F` instead of `&[F]`? Is the idea that in the future, we might support vectors of query randomness in some other PCP?

Reply:
Yeah, that's the idea. I don't foresee it happening, but I figured it's a good idea to be general in the API.